Add Form setting to Zapier step

richardW8k opened the pull request:
## Description
This restores the ability to use the Zapier step with forms which don’t have Zapier feeds. It also adds a Form setting to the Zapier step settings so the user can select a different form as the source of the feeds to be processed by the step.

Zapier Add-On 4.1 will add support for Zapier storing the zap ID in the feed when created or updated. If the feed meta contains the `zapID` a link to view the zap on zapier.com will be included in a tooltip next to the feed name.

![Screenshot 2021-05-05 at 18 44 22](https://user-images.githubusercontent.com/1872371/117190131-efa35f00-add6-11eb-8ad6-bd77d973372a.png)


## Testing instructions
- Enable logging for the Zapier add-on
- Create a form with a text field
- Duplicate the form
- Enable display of feeds on the Forms > Settings > Zapier page, if not already enabled
- Using an API client such as Postman or Paw create a `POST` request for the `/wp-json/gf/v2/feeds` endpoint of your site. Use basic auth credentials created via the Forms > Settings > REST API page. Use JSON for the request body with the following config, replacing 180 with the ID of the first form:
```
{
  "form_id": "180",
  "addon_slug": "gravityformszapier",
  "meta": {
    "feedName": "rest api test",
    "zapURL": "https://testing.test",
    "adminLabels": true,
    "zapID": "123"
  }
}
```
![Screenshot 2021-04-28 at 14 08 46](https://user-images.githubusercontent.com/1872371/116408954-4931e980-a82b-11eb-8f7c-33a0370781f3.png)

- Send the request
- Create a Zapier step for the second form
- Select the first form in the Form setting
- Select the "rest api test" feed.
- Confirm the tooltip contains a link to view the zap on zapier.com. Note: while the link will open zapier in a new tab it won't display the zap editor since we are testing with an invalid id.
- Save the feed
- Preview the second form
- Add a value to the field and submit
- View the logs to find the Zapier feed from the first form was used to process the second form
- Create a Zapier feed for the second form, use a different feed name 
- Edit the Zapier step, change the Form setting to the "This Form" choice, and select the new feed
- Submit the form again
- Check the logs to confirm the correct Zapier feed was used

## Automated Test Enhancements
N/A

## Documentation Changes
N/A

## Checklist:
- [x] I've tested the code.
- [x] My code follows the WordPress code style. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code follows the inline documentation standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/ -->

saifsultanc commented:
On the step (fourth from last), `Create a Zapier feed for the second form, use a different feed name`, there seems to be something bugged. Unable to create a new Zapier feed, since it doesn't let me fill in the feed details.

<img width="532" alt="Screenshot 2021-05-04 at 6 38 47 PM" src="https://user-images.githubusercontent.com/26293394/117008271-29f1fb00-ad08-11eb-8a3f-0de767272641.png">

richardW8k commented:
@saifsultanc sorry, i should have mentioned, you need to use the rest api call again. manually creating the feeds via the ui has been disabled, the add new buttons have been removed in the 4.1 release which should be going out later today

saifsultanc reviewed (approved):
Tested again with the rest api call for the feed generation on the 'second' form. Looks fine.

richardW8k commented:
@saifsultanc can you check the feed tooltip on the zapier step again? I've updated it to include links to view and edit the zap and added translator comments. Zapier 4.1 has now been released so if you want you can create a zap on zapier.com to test a real feed with a valid zap id.
